### PR TITLE
fix: presist branch when adding files and refresh commits on branch switch

### PR DESCRIPTION
**Pull request by pawanpaudel93**
# Fixes

- Fixed issue related to switching branch when adding files in a branch.
- Fixed issue related to not fetching commits when switching branch.

### CODE REVIEW

**Review by kranthicodes — APPROVED** [review]
tACK. 
.works as expected. Thanks for working on this.

**Comment by kranthicodes** [comment]
@t8 do you want to take second review or we are good?

ps: @pawanpaudel93 Please write a summary that describes the problem this PR is addressing and your proposed solution. 

```
## Summary
<describe the PR in detail>

## Test
If you did any unit testing or manual testing and everything works as expected

## Screenshots
Short video or screenshot of how solution is working by using the UI.
```

Really helps the reviewer.

**Review by t8 — APPROVED** [review]
LGTM!

**Pull request merged**
